emily/handler/src/api/models/deposit/requests.rs

```diff
@@ -43,6 +53,82 @@ pub struct CreateDepositRequestBody {
     pub reclaim_script: String,
     /// Deposit script.
     pub deposit_script: String,
+    /// The raw transaction hex.
+    pub transaction_hex: String,
+}
+
+/// This is the dust limit for deposits in the sBTC smart contracts.
+/// Deposit amounts that is less than this amount will be rejected by the
+/// smart contract.
+pub const DEPOSIT_DUST_LIMIT: u64 = 546;
+
+fn parse_hex<T, F, E>(input: &str, error_msg: &str, parser: F) -> Result<T, Error>
+where
+    F: Fn(&str) -> Result<T, E>,
+{
+    parser(input).map_err(|_| Error::HttpRequest(StatusCode::BAD_REQUEST, error_msg.to_string()))
+}
+
+impl CreateDepositRequestBody {
+    /// Validates that the deposit request is valid.
+    /// This includes validating the request fields, if their content matches the transaction
+    /// and if the amount is within the limits.
+    pub fn validate(&self, limits: &Limits) -> Result<DepositInfo, Error> {
+        let deposit_req = CreateDepositRequest {
+            outpoint: OutPoint {
+                txid: parse_hex(&self.bitcoin_txid, "invalid bitcoin_txid", Txid::from_str)?,
+                vout: self.bitcoin_tx_output_index,
+            },
+            reclaim_script: parse_hex(
+                &self.reclaim_script,
+                "invalid reclaim_script",
+                ScriptBuf::from_hex,
+            )?,
+            deposit_script: parse_hex(
+                &self.deposit_script,
+                "invalid deposit_script",
+                ScriptBuf::from_hex,
+            )?,
+        };
+
+        let tx: Transaction = parse_hex(
+            &self.transaction_hex,
+            "invalid transaction_hex",
+            encode::deserialize_hex,
+        )?;
+
+        let amount = tx
+            .tx_out(self.bitcoin_tx_output_index as usize)
+            .map_err(|_| {
+                Error::HttpRequest(
+                    StatusCode::BAD_REQUEST,
+                    "invalid bitcoin_output_index".to_string(),
+                )
+            })?
+            .value
+            .to_sat();
```

Comment:
nit: you can remove this if you move up the `validate_tx` call, which I think also has a dedicated error message for this check:
```suggestion
        let deposit_info = deposit_req
            .validate_tx(&tx)
            .map_err(|e| Error::HttpRequest(StatusCode::BAD_REQUEST, e.to_string()))?;
```
Then below just do
```rust
...
if deposit_info.amount < min {
    ...
```

Reply:
I noticed this, but I preferred the current approach because I wanted the deposit's amount to be validated before the transaction matching, and I found the error message for the existing validation (`could not get outpoint f75cb869600c6a75ab90c872435da38d54d53c27afe5e03ac7dedae7822958de:10 from BTC transaction: invalid output index`) too verbose. Said that, I am aware that my motivations are just a small subjective preference, so I am open to change if you think the other approach would be clearer!

Reply:
Oh okay, no worries. What is here is fine then.